src/Documenter.jl

```diff
@@ -750,4 +758,85 @@ function generate(pkgname::AbstractString; dir=nothing)
     nothing
 end
 
+const external_dir = "DOCUMENTER_EXTERNAL"
```

Comment:
I might actually put these pages right under `src/` by default.

Furthermore, the main issue with these "external" pages is that Documenter does not know what the name (i.e. the URL) of the page should be. For the normal pages it is determined from the path (e.g. `src/dir/page.md` becomes `/dir/page/`).

So, I would actually propose `external(src, target)`. E.g. `external("../README.md", "info/readme.md")`. And `target` could be omitted, with default being the `src` root and the name of the file (i.e. `dirname(src)`).

I'm not sure what to do with relative links in the source files. E.g. say there's a link to an authors file in the readme: `[authors file](AUTHORS.md)`. We could try being smart about resolving these links (e.g. if `../README.md` goes to `/readme/` and `../AUTHORS.md` goes to `/pages/authors/`, then they would still resolve to each other). But for now, I guess the easiest thing would be that it's up to the user to make sure that the page structure is replicated with the `external` calls.

Reply:
I took your advice and added a `target` keyword, and by default they're put directly in `src`.

As for relative links, I added a warning in the `external` docstring, and we can leave the rest up to the user.